src/utils/codec.ts

```diff
@@ -0,0 +1,154 @@
+/* eslint-disable no-bitwise */
+/**
+ * @file Codec
+ * @name codec.ts<utils>
+ * @description |
+ *
+ *    |    base    |  bitwise  |  bits  |
+ *    | ---------- | --------- | ------ |
+ *    |    16      |  1 << 4   |   4    |
+ *    |    64      |  1 << 6   |   6    |
+ *    |    256     |  1 << 8   |   8    |
+ *    |    1024    |  1 << 10  |   10   |
+ *
+ */
+interface CodecSchema {
+    bits: number
+    table: string[]
+}
```

Comment:
[Base4096Hangul Format][#5] can be implemented by just adding a schema to this 'codec framework'.

[#5]: https://github.com/DimensionDev/Tessercube2-Meta-DSD/issues/5